src/views/Settings.vue

```diff
@@ -0,0 +1,72 @@
+<template>
+  <div v-if="hasInitialFetchDone" :class="$style.container">
+    <mobile-setting-modal v-if="isMobile">
+      <router-view />
+    </mobile-setting-modal>
+    <desktop-setting-modal v-else>
+      <router-view />
+    </desktop-setting-modal>
+  </div>
+  <div v-else></div>
+</template>
+
+<script lang="ts">
+import { defineComponent, Ref, ref, watch, computed } from 'vue'
+import { onBeforeRouteLeave, useRoute, useRouter } from 'vue-router'
+import { RouteName } from '@/router'
+import { defaultSettingsName } from '@/router/settings'
+import useIsMobile from '@/use/isMobile'
+import useInitialFetch from './use/initialFetch'
+import DesktopSettingModal from '@/components/Settings/DesktopSetting.vue'
+import MobileSettingModal from '@/components/Settings/MobileSetting.vue'
+import store from '@/store'
+
+const useSettingsRootPathWathcer = (isMobile: Ref<boolean>) => {
+  const route = useRoute()
+  const router = useRouter()
+  const redirectOrMarkRootIfNeeded = () => {
+    if (route.name !== RouteName.Settings) {
+      return
+    }
+    if (isMobile.value) {
+      store.commit.ui.settings.setSettingsRootShown(true)
+    } else {
+      router.replace({ name: defaultSettingsName })
+    }
+  }
+  watch([computed(() => route.name), isMobile], redirectOrMarkRootIfNeeded, {
```

Comment:
```suggestion
  watch([toRef(route, 'name'), isMobile], redirectOrMarkRootIfNeeded, {
```
内部実装こっちのが簡単っぽいのでこっちのがよさそうです(実際どうなのかはわかってないです…)